File: Objects/Entities/LaunchBlock.cs
using Celeste.Mod.Entities;
using Microsoft.Xna.Framework;
using Monocle;
using MonoMod.Utils;

namespace Celeste.Mod.legendofthemonsterrealm.Objects.Entities
{
    [CustomEntity("legendofthemonsterrealm/LaunchBlock")]
    public class LaunchBlock : Solid
    {
        private readonly DynamicData BaseData;

        public LaunchBlock(EntityData data, Vector2 offset) : base(data.Position + offset, data.Width,data.Height,false)
        {
            BaseData = DynamicData.For(this);
            // other stuff
        }

        public override void Update()
        {
            Player player = GetPlayerOnTop();
            if(player!=null&&player.Speed.Y>=0)
            {
                player.Speed = new Vector2(player.Speed.X*2, -50);
            }
            base.Update();
        }
        public override void Render()
        {
            Draw.Rect(Collider, Color.Purple);
        }
    }
}

File: Loenn/entities/LaunchBlock.lua
local utils = require("utils")

local LaunchBlock = {}

LaunchBlock.name = "legendofthemonsterrealm/LaunchBlock"
LaunchBlock.color = {0.8, 0.4, 0.4, 0.8}
LaunchBlock.canResize = {true, true}
LaunchBlock.placements = {
    name = "legendofthemonsterrealm/LaunchBlock",
    data = {
        width = 8,
		height = 8,
    }
}

function LaunchBlock.rectangle(room, entity)
    return utils.rectangle(entity.x, entity.y, entity.width or 8, entity.height or 8)
end

return LaunchBlock